### PR TITLE
"audio or video avoids automatically playing audio" (80f0bf / aaa1bf / 4c31df): Feedback from TF.

### LINKED ISSUE #1016: auto-play rule uses ambiguous definition "mechanism" [4c31df]
The rule [https://act-rules.github.io/rules/4c31df](https://act-rules.github.io/rules/4c31df) uses the WCAG definition of "mechanism". This definition is somewhat ambiguous, and so should be replaced with something unambiguous.

### PR DESCRIPTION
**Pull request by Jym77**
* Add definition of "attribute value" for IDL attributes, and use it in the rules.
* Replace formulation of  source using "media resource".
* Replace "mechanism" by "instrument".
* Add an assumption that alternance of short sounds and pauses is not good.
* Add assumption that the mechanism must be located in the same page, and exposed to users.

Closes issue(s):

- closes #https://github.com/w3c/wcag-act/issues/456
- closes #1016 

Need for Final Call:
This will require a 1 week Final Call (large editorial changes and clarifications).

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by kasperisager — APPROVED** [review]

File: _rules/auto-play-audio-has-control-mechanism-4c31df.md [hunk 1]
```diff
@@ -139,7 +137,7 @@ This `video` element autoplays and does not have a [mechanism](https://www.w3.or
 
 #### Failed Example 3
 
-This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to pause, stop, or turn the audio volume off but the mechanism is not visible.
+This `video` element has a [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not visible.
```
Comment (WilcoFiers) [inline _rules/auto-play-audio-has-control-mechanism-4c31df.md:140]:
```suggestion
This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not visible.
```

File: _rules/auto-play-audio-has-control-mechanism-4c31df.md [hunk 2]
```diff
@@ -175,7 +173,7 @@ This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechani
 
 #### Failed Example 4
 
-This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to pause, stop, or turn the audio volume off but its `button` elements do not have accessible names.
+This `video` element has a [instrument][] to pause, stop, or turn the audio volume off but its `button` elements do not have accessible names.
```
Comment (WilcoFiers) [inline _rules/auto-play-audio-has-control-mechanism-4c31df.md:176]:
```suggestion
This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but its `button` elements do not have accessible names.
```

File: _rules/auto-play-audio-has-control-mechanism-4c31df.md [hunk 3]
```diff
@@ -210,7 +208,7 @@ This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechani
 
 #### Failed Example 5
 
-This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to pause, stop, or turn the audio volume off but the mechanism is not included in the accessibility tree.
+This `video` element has a [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not included in the accessibility tree.
```
Comment (WilcoFiers) [inline _rules/auto-play-audio-has-control-mechanism-4c31df.md:211]:
```suggestion
This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not included in the accessibility tree.
```

**Review by WilcoFiers — DISMISSED** [review]
Editorial stuff

File: _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md [hunk 4]
```diff
@@ -43,23 +48,23 @@ htmlHintIgnore:
...
 
-- an `autoplay` [attribute value][] of `true`, and
-- a `muted` [attribute value][] of `false`, and
-- a `paused` attribute whose value is false, and
-- either a `src` attribute or a child `source` element that references content with a duration of more than 3 seconds that contains audio.
-
-**Note**: [`autoplay`](https://html.spec.whatwg.org/multipage/media.html#attr-media-autoplay) and [`muted`](https://html.spec.whatwg.org/multipage/media.html#attr-media-muted) are both HTML attributes set on elements. On the other hand, [`paused`](https://html.spec.whatwg.org/multipage/media.html#dom-media-paused) is a DOM attribute which is not specified in HTML.
+- an `autoplay` [attribute value][] of `true`; and
+- a `muted` [attribute value][] of `false`; and
+- a `paused` [attribute value][] of `false`; and
```
[the remark this answers is not in the document]
Comment (kasperisager) [reply _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md:53]: It refers to an IDL attribute, not a content attribute, as covered in the definition of "attribute value":

> The attribute value of an IDL attribute is the value returned on getting it. Note that when an IDL attribute reflects a content attribute, they have the same attribute value.

[the remark this answers is not in the document]
Comment (Jym77) [reply _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md]: Which is why the definition of "attribute value" has been updated in that very PR in order to handle IDL attributes…

File: _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md [hunk 5]
```diff
@@ -41,25 +46,25 @@ htmlHintIgnore:
...
+- This rule assumes that the [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to control the sound must be located in the same [web page][]. Note that mechanisms that are provided by user agents, assistive technologies, or operating systems are hardly testable as they depend on the user. Note that mechanisms that are located on other pages can still create accessibility issues for users relying on sound to navigate (e.g. screen readers users) since the autoplaying sound will interfere with their ability to find and activate the mechanism. If a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) external to the [web page][] is provided, it is possible to fail this rule but still satisfy [Success Criterion 1.4.2 Audio Control][sc142].
+- This rule assumes that the [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to control the sound must be visible and accessible in order to be effective and usable by all kind of users. If the mechanism is hidden to some users, it is possible to fail this rule but still satisfy [Success Criterion 1.4.2 Audio Control][sc142].
```
Comment (daniel-montalvo) [inline _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md:67]:
```suggestion
- This rule assumes that the [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to control the sound must be visible and accessible in order to be effective and usable by all kinds of users. If the mechanism is hidden to some users, it is possible to fail this rule but still satisfy [Success Criterion 1.4.2 Audio Control][sc142].
```

File: _rules/auto-play-audio-has-control-mechanism-4c31df.md [hunk 6]
```diff
@@ -139,7 +137,7 @@ This `video` element autoplays and does not have a [mechanism](https://www.w3.or
 
 #### Failed Example 3
 
-This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to pause, stop, or turn the audio volume off but the mechanism is not visible.
+This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not visible.
```
Comment (daniel-montalvo) [inline _rules/auto-play-audio-has-control-mechanism-4c31df.md:140]:
```suggestion
This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the [instrument][] is not visible.
```

File: _rules/auto-play-audio-has-control-mechanism-4c31df.md [hunk 7]
```diff
@@ -210,7 +208,7 @@ This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechani
 
 #### Failed Example 5
 
-This `video` element has a [mechanism](https://www.w3.org/TR/WCAG21/#dfn-mechanism) to pause, stop, or turn the audio volume off but the mechanism is not included in the accessibility tree.
+This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the mechanism is not included in the accessibility tree.
```
Comment (daniel-montalvo) [inline _rules/auto-play-audio-has-control-mechanism-4c31df.md:211]:
```suggestion
This `video` element has an [instrument][] to pause, stop, or turn the audio volume off but the [instrument][] is not included in the accessibility tree.
```

**Review by daniel-montalvo — CHANGES_REQUESTED** [review]
Minor editorial suggestions

**Review by daniel-montalvo — APPROVED** [review]

File: _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (WilcoFiers) [reply _rules/audio-or-video-avoids-automatically-playing-audio-80f0bf.md]: Fair enough! My mistake.

**Review by WilcoFiers — APPROVED** [review]

**Comment by Jym77** [comment]
Final Call ends on October 1st.

**Comment by Jym77** [comment]
Final Call has ended. Merging.

**Pull request merged**